core/math/vector2.h

```diff
@@ -195,6 +195,9 @@ struct [[nodiscard]] Vector2 {
 			x(0), y(0) {}
 	constexpr Vector2(real_t p_x, real_t p_y) :
 			x(p_x), y(p_y) {}
+	template <typename V, std::enable_if_t<std::is_arithmetic_v<V>, int> = 0>
+	constexpr explicit Vector2(V p_v) :
```

Comment:
Making this a template argument is a bit overambitious. 
What `Vector` needs for `x` and `y` is `real_t`. It copies the `real_t` to `x` and `y`. So the argument should be `real_t`.

Reply:
It conflicts with the copy constructor Vector2(Vector2 p_v) that is generated by the compiler, we will have to redefine it for all types or the compiler will be confused when we use Variant. See how it was handled in https://github.com/godotengine/godot/pull/104404 to avoid this issue.

Reply:
Right... The appropriate fix would be to remove implicit conversions that cause the conflict. Though that's out of scope for this PR.